src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -202,7 +205,13 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
                 if (targetRouteBuilder is null)
                 {
                     // The app defined endpoints without calling UseRouting() explicitly, so call UseRouting() implicitly.
-                    app.UseRouting();
+                    app.UseRouting(overrideEndpointRouteBuilder: false);
```

Comment:
Why do we need to call this overload? We know there's no IEndpointRouteBuilder in this case, right?

Reply:
Not quite. `UseExceptionHandler` is added by the user in Program.cs which occurs before app.Run() which is where this configuration runs. So UseRouting() here is configuring an EndpointRoutingMiddleware before the ExceptionHandlerMiddleware in the pipeline but doing so after UseExceptionHandler() is called.

Let me know if I need to be clearer since it's all a bit magical.

Reply:
Can you provide a test case there's already an `IEndpointRouteBuilder` in `app.Properties` here? It's not easy for code in Program.cs to get access to this `app` instance. Code in Program.cs is modifying `_builtApplication`, so any call to `UseExceptionHandler` would happen later where we call `_builtApplication.BuildRequestDelegate()`.

The only thing I could think of is an `IStartupFilter` or something like that. If we really care about a startup filter calling `app.UseRouting()` before calling `next(app)`, we could skip calling `UseRouting()` altogether. I'm not convinced that's realistic though, so I'm not sure it's worth complicating the code for that case for this. If we do, we should add a regression test.

Reply:
Hmm now I think about it, I don't recall why I set the argument to false here. However, I do clearly recall without it, things didn't work. I could debug it to come up with a more clear explanation.

That being said, after chatting with @javiercn today, I'm not sure if this is the approach we are going to take so I think it's a moot point. I think there's more investigation/design to be done before addressing this specific feedback.